app/api/logging/audit.py

```diff
@@ -20,54 +20,117 @@
+        # be audited for each thread.
+        "sys.settrace": (),
+        # Detect monkey patching of types and objects. This event is raised for
+        # the __class__ attribute and any attribute on type objects.
+        "object.__setattr__": ("object", "attr", "value"),
+        # Detect deletion of object attributes. This event is raised for any
+        # attribute on type objects.
+        "object.__delattr__": ("object", "attr"),
+        # Detect imports and global name lookup when unpickling.
+        "pickle.find_class": ("module_name", "global_name"),
+        # Notifies hooks they are being cleaned up, mainly in case the event is
+        # triggered unexpectedly. This event cannot be aborted.
+        "sys._clearaudithooks": (),
+        # Detect dynamic creation of code objects. This only occurs for direct
+        # instantiation, and is not raised for normal compilation.
+        "code.__new__": ("bytecode", "filename", "name"),
+        # Detect dynamic creation of function objects. This only occurs for
+        # direct instantiation, and is not raised for normal compilation.
+        "function.__new__": ("code",),
+        # Detect when native modules are used.
+        "ctypes.dlopen": ("module_or_path",),
+        # Collect information about specific symbols retrieved from native modules.
+        "ctypes.dlsym": ("lib_object", "name"),
+        # Detect when code is accessing arbitrary memory using ctypes.
+        "ctypes.cdata": ("ptr_as_int",),
+        # Detects creation of mmap objects. On POSIX, access may have been
+        # calculated from the prot and flags arguments.
+        "mmap.__new__": ("fileno", "map_size", "access", "offset"),
+        # Detect when code is accessing frames directly.
+        "sys._current_frames": (),
+        # Detect access to network resources. The address is unmodified from the original call.
+        "socket.address": ("socket", "address"),
+        # Detects URL requests.
+        "urllib.Request": ("url", "data", "headers", "method"),
+    }
+
+    if event_name not in EVENTS_TO_LOG:
         return
 
-    audit_log(event_name, args)
+    arg_names = EVENTS_TO_LOG[event_name]
+    log_audit_event(event_name, args, arg_names)
 
 
-def audit_log(event_name: str, args: tuple[Any, ...]) -> None:
+def log_audit_event(event_name: str, args: Sequence[Any], arg_names: Sequence[str]) -> None:
     """Log a message but only log recently repeated messages at intervals."""
-    key = (event_name, repr(args))
-    count = audit_message_count[key] = audit_message_count[key] + 1
-    if count <= 10 or (count <= 100 and (count % 10) == 0) or (count % 100) == 0:
-        extra = {"audit.event_name": event_name, "count": count}
-        for arg_index, arg in enumerate(args):
-            extra["audit.args.%i" % arg_index] = arg
-        logger.log(AUDIT, event_name, extra=extra)
+    arg_keys = map(lambda arg_name: f"audit.args.{arg_name}", arg_names)
+    extra = dict(zip(arg_keys, args))
```

Comment:
Could be combined into one dict comprehension:

```suggestion
    extra = {f"audit.args.{arg_name}": arg for arg_name, arg in zip(arg_names, args)}
```